feature-libs/product-configurator/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.ts

```diff
@@ -47,7 +47,7 @@ export interface ConfiguratorAttributeProductCardComponentOptions {
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ConfiguratorAttributeProductCardComponent
-  extends ConfiguratorAttributeBaseComponent
+  extends ConfiguratorUiKeyGeneratorService
```

Comment:
why this one extends a service?

Reply:
Yes, we will handle it accordingly

Reply:
Done! `ConfiguratorUiKeyGeneratorService` will be injected via constructor